Added ADAM6250 Device Class

Patfra opened the pull request:
Closes #4

1. Working ADAM6250 Device Class with attributes:

 * DigitalOutput(bool) 
 * Counter(bool)  
 * LatchStatus(bool)
 * ClearOverflow(bool) 
 * Counter/Frequency(int) 
 * DigitalOutput(bool) 
 * Pulse Output Low(int) 
 * Pulse Output High(int)
 * AbsolutePulse(int)
 * IncrementalPulse(int)

2. Sphinx documentation

Patfra committed:
#3 Minor changes
- added description to commands
- adapted __init__.py and run_server.py to run third Device Class

Patfra committed:
#4 Added working ADAM-6250 Device Class

Patfra committed:
#4 Added Sphinx documentation files

LJBD commented on adam/__init__.py:
```diff
@@ -4,17 +4,19 @@
-- ADAM6224 device class - ADAM6224 is a 4-ch  Isolated Analog Output Modbus TCP Module with additional 4-ch Digital Input
-- ADAM6250 device class - ADAM6250 is a 15-ch  Isolated Digital I/O Modbus TCP Module
+* ADAM6224 device class - ADAM6217 is a 8-ch  Isolated Analog Input Modbus
+TCP Module
```
Indentation!

LJBD commented on adam/__init__.py:
```diff
@@ -4,17 +4,19 @@
+* ADAM6224 device class - ADAM6217 is a 8-ch  Isolated Analog Input Modbus
+TCP Module
+* ADAM6224 device class - ADAM6224 is a 4-ch  Isolated Analog Output Modbus
+TCP Module with additional 4-ch Digital Input
```
Indentation!

LJBD commented on adam/__init__.py:
```diff
@@ -4,17 +4,19 @@
+* ADAM6224 device class - ADAM6224 is a 4-ch  Isolated Analog Output Modbus
+TCP Module with additional 4-ch Digital Input
+* ADAM6250 device class - ADAM6250 is a 15-ch  Isolated Digital I/O Modbus
+TCP Module
```
Indentation!

LJBD commented on adam/adam_6217.py:
```diff
@@ -411,11 +411,13 @@ class ADAM6217(Device):
         self.set_status("ADAM-6217 enabled")
 
     def delete_device(self):
+        """Delete device"""
```
This doesn't reflect what this method calls.

LJBD commented on adam/adam_6217.py:
```diff
@@ -411,11 +411,13 @@ class ADAM6217(Device):
     # ---------------
 
     def init_device(self):
+        """Initiate device and sets its state to STANDBY"""
```
*Initialise

LJBD commented on adam/adam_6250.py:
```diff
@@ -0,0 +1,352 @@
+    # ---------------
+
+    def init_device(self):
+        """Initiate device and sets its state to STANDBY"""
```
*Initialise

LJBD commented on adam/adam_6250.py:
```diff
@@ -0,0 +1,352 @@
+            "device")
+
+    def delete_device(self):
+        """Delete device"""
```
Please make this more verbose - add a message that it disconnects the ADAM.

Patfra replied:
Will that be appropriate?
```python      
""" Disconnect from device and sets state to STANDBY """
        self.connected_ADAM.close()
        self.set_state(DevState.STANDBY)
        self.set_status(
            "Device disconnected form ADAM-6250, set state to STANDBY, "
            "ready to connect to device again")
```

LJBD replied:
Well, if you delete the device, you don't need to set its state and status, because it is going to be deleted when this method finishes!
I was thinking only of updating the docstring.

Patfra committed:
#4 Added requested changes
- fixed missing indentation
- fixed spelling mistakes
- upgraded delete_device command
- added better description to delete_device command
- fixed minor mistakes in code

LJBD commented on adam/adam_6217.py:
```diff
@@ -411,12 +411,18 @@ class ADAM6217(Device):
     def delete_device(self):
+        """Disconnect from device and sets state to STANDBY """
         self.connected_ADAM.close()
+        self.set_state(DevState.STANDBY)
```
Setting state and status isn't necessary if you're deleting a device.

LJBD commented on adam/adam_6224.py:
```diff
@@ -359,13 +363,19 @@ def write_TypeCode(self, value, channel):
     def delete_device(self):
+        """Disconnect from device and sets state to STANDBY """
         self.connected_ADAM.close()
+        self.set_state(DevState.STANDBY)
```
You don't need to set the state and status when you're deleting a device!

LJBD commented on adam/adam_6250.py:
```diff
@@ -0,0 +1,363 @@
+    def delete_device(self):
+        """Disconnect from device and sets state to STANDBY """
+        self.connected_ADAM.close()
+        self.set_state(DevState.STANDBY)
```
Also here. Deleting a device means it won't have any state and status after this operation, so it's pointless.

Patfra replied:
That would be ok?
```python
    def delete_device(self):
        """Disconnect from physical device before deleting instance"""
        self.connected_ADAM.close()


    @command
    @DebugIt()
    def disconnect(self):
        """Disconnect from device and sets state to STANDBY """
        self.connected_ADAM.close()
        self.set_state(DevState.STANDBY)
        self.set_status(
            "Device disconnected form ADAM-6250, set state to STANDBY, "
            "ready to connect to device again")

```

LJBD replied:
Yes.

Patfra committed:
#4 Added requested changes
- changed delete_device command
- added disconnect command
